contributions/api/controllers/contributions.py

```diff
@@ -382,43 +392,46 @@ def capabilities_search(id):
 
     return capability_dataset
 
-def alltalentssearch():
+def alltalentssearch(name=None):
     args = request.args
     query = dict()
     is_list = False
 
     if len(args) == 0:
         is_list = True
 
-    try:
-        query = query_params.format_query_talent(args, query)
-    except Exception as ex:
-        msg = {
-            "reason": "The query is wrong or bad argument " + str(args),
-            "error": "Bad Request: " + request.url,
-        }
-        msg_json = jsonutils.create_log_json("Talent", "SEARCH", msg)
-        logging.error("Talent SEARCH " + json.dumps(msg_json))
-        return rs_handlers.bad_request(msg_json)
+    if is_list:
+        out_json = mongoutils.get_result(coll_contribution, query)
+    else:
+        try:
+            query = query_params.format_query_talent(name, query)
```

Comment:
do we have the `name=None` all the time? I am trying to understand the code so just asking for clarity

Reply:
I believe so. We have to have this to directly get the name variable from query string.

Reply:
It's set to None as this is an optional parameter. For required parameter (query or path), the default None value won't be there.